sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_async_lro.py

```diff
@@ -76,6 +77,101 @@ async def _poll(self):  # pylint:disable=invalid-overridden-method
             )
 
+class AnalyzeHealthcareEntitiesAsyncLROPollingMethod(TextAnalyticsAsyncLROPollingMethod):
+
+    def __init__(self, *args, **kwargs):
+        self._text_analytics_client = kwargs.pop("text_analytics_client")
+        super(AnalyzeHealthcareEntitiesAsyncLROPollingMethod, self).__init__(*args, **kwargs)
+
+    @property
+    def _current_body(self):
+        from ._generated.v3_1_preview_3.models import JobMetadata
+        return JobMetadata.deserialize(self._pipeline_response)
+
+    @property
+    def created_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.created_date_time
+
+    @property
+    def expires_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.expiration_date_time
+
+    @property
+    def last_modified_on(self):
+        if not self._current_body:
+            return None
+        return self._current_body.last_update_date_time
+
+    @property
+    def id(self):
+        if not self._current_body:
+            return None
+        return self._current_body.job_id
+
+
+class AnalyzeHealthcareEntitiesAsyncLROPoller(AsyncLROPoller[PollingReturnType]):
+
+    @property
+    def created_on(self):
+        return self._polling_method.created_on
+
+    @property
+    def expires_on(self):
+        return self._polling_method.expires_on
+
+    @property
+    def last_modified_on(self):
+        return self._polling_method.last_modified_on
+
+    @property
+    def id(self):
+        return self._polling_method.id
+
+    async def cancel( # type: ignore
+        self,
+        **kwargs
+    ):
+        """Cancel the operation currently being polled.
+
+        :keyword int polling_interval: The polling interval to use to poll the cancellation status.
+            The default value is 5 seconds.
+        :return: Returns an instance of an LROPoller that returns None.
+        :rtype: ~azure.core.polling.LROPoller[None]
+        :raises: Warning when the operation has already reached a terminal state.
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/async_samples/sample_analyze_healthcare_entities_with_cancellation_async.py
+                :start-after: [START analyze_healthcare_entities_with_cancellation_async]
+                :end-before: [END analyze_healthcare_entities_with_cancellation_async]
+                :language: python
+                :dedent: 8
+                :caption: Cancel an existing health operation.
+        """
+        polling_interval = kwargs.pop("polling_interval", 5)
+
+        terminal_states = ["cancelled", "cancelling", "failed", "succeeded", "partiallyCompleted", "rejected"]
+        await self._polling_method.update_status()
+
+        if self._polling_method.status() in terminal_states:
+            raise Warning("Operation with ID '%s' is already in a terminal state and cannot be cancelled." \
```

Comment:
@abhahn the more I think about it, the more I think we should not check the status, and just make the network call. We try to do as little client side validation as possible